npc_simulator/CMakeLists.txt

```diff
@@ -1,92 +1,65 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(npc_simulator)
 
-add_compile_options(-std=c++14)
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
+  # TODO remove
+  add_compile_options(-Werror)
+endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  message_generation
-  roscpp
-  sensor_msgs
-  std_msgs
-  tf2
-  tf2_geometry_msgs
-  tf2_ros
-  autoware_perception_msgs
-  autoware_lanelet2_msgs
-  uuid_msgs
-  unique_id
-  lanelet2_extension
-  dummy_perception_publisher
-)
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
-add_message_files(
-  DIRECTORY msg
-  FILES
-    LaneChangeDir.msg
-    LaneFollowMode.msg
-    Object.msg
-)
+# message generation
 
-add_service_files(
-  DIRECTORY srv
-  FILES
-    GetObject.srv
-)
+# doesn't work with ament_auto when interface and client code in same package
```

Comment:
Actually it doesn't work with ament_auto at all, not just in the same package with the code

Reply:
I removed those comments, they were more for myself while getting it t work